libraries/AP_HAL_Linux/Util.h

```diff
@@ -48,7 +48,7 @@ class Util : public AP_HAL::Util {
      */
     void set_system_clock(uint64_t time_utc_usec);
     const char* get_custom_log_directory() { return custom_log_directory; }
-    const char* get_custom_terrain_directory() { return custom_terrain_directory; }
+    const char* get_custom_terrain_directory() const final { return custom_terrain_directory; }
```

Comment:
> Exactly. I wanted to demonstrate the it detects the same kind of bugs that override is supposed to detect.

You realize that only worked because the method wasn't marked virtual right? Which isn't usually the case. I actually prefer that overriding methods are marked virtual as well, to make devs aware of its status.